GSOC18/GeometryUtilities-master/interpolation/hexCells_to_squareCell_interpolation.py

```diff
@@ -51,15 +56,23 @@ def linear_interpolate_hex_to_square(hex_cells_dict,layer,resolution=(500,500)):
 
     #Iterating over all the cells to get the bounds of the detector
     print '>>> Calculating Bounds'
-    center_x=map(lambda c:c.center.x,cells_dict.values())
-    max_x=max(center_x)
-    min_x=min(center_x)
-    center_y=map(lambda c:c.center.y,cells_dict.values())
-    max_y=max(center_y)
-    min_y=min(center_y)
+    bounds_cords=map(lambda c:c.vertices.bounds,cells_dict.values())
```

Comment:
Bug Fix Is mainly done here.

Reply:
Earlier I was calculating the bounds based on the center of the hexagonal cells not the actual bounds of cells.
Now the actual bound if found and around it the square mesh is created.

The actual problem was that at the edges some of the cells were not completely inside the square mesh, hence their areas were not counted in the overlap coef but when creating the image the complete area of those cells were distributed and hence the slight increase in their total area there at the edges.

Thanks a lot for pointing out otherwise I wouldn't have rechecked it.